### DISCUSSION TITLE
troubleshooting owncast-demobot & emojiwall

### DISCUSSION THREAD

**Discussion by meljoann** [owncast/owncast#3185] [Q&A] [answered]
Hi everyone, I'm trying to set up some simple bots and having trouble. I'm running a reverse proxy in nginx.

# owncast-demobot
-  my only customisations so far are custom text
- bot receives events from both curl and Owncast chat messages, with the "Tracing request for..." output showing up
- Owncast error logs are always "invalid access token"

## what I've tried so far:
- regenerated access token from admin panel
- the correct access token is in config.yaml
- tried starting the docker container with environment variables containing the access token
- added in nginx:
```
	location /auto {
		proxy_pass http://127.0.0.1:8100;
		proxy_buffering off;
		proxy_http_version 1.1;
            proxy_set_header Upgrade $http_upgrade;
            proxy_set_header Connection "upgrade";
```

# emojiwall
- could be a related problem?
- using the https://smol.stream/ overlay
- the emojis display correctly using curl
- emojis don't display when I put the emojis in the Owncast chat -- they use a "broken image" icon

Very new to this so may have missed something obvious...?

Thanks!
Mel

**Comment by gabek** [comment]
Oh boy, I didn't build either one of those tools but I'll try and help debug!

For the emojiwall's broken images, what is the URL it's trying to access?

For the demobot, these are the two environment variables set in the config file that I'm using, I just want to make sure they match with yours:

`OWNCAST_DEMOBOT_ACCESSTOKEN` and `OWNCAST_DEMOBOT_OWNCASTADDRESS`.

You mentioned a config file, but I'm not actually using a config file myself, just those two environment variables. It was so long ago I set it up I don't remember if it supports both a config file and the environment? But just in case maybe try using environment variables instead?

**Comment by gabek** [comment, marked as answer]
Oh one other thing, and I don't know if I'm just misreading this but in your config above it looks like you're routing `/auto` to `/`. I don't remember the specifics but should that be routing `/` to `/` so `/auto` routes to `/auto`? Otherwise it would have to access `/auto/auto` to hit the `/auto` route on the bot, no?

**Reply by meljoann** [reply]
Awesome, thanks so much for this Gabe! I'll try again with these clues tomorrow, and report back :)

**Reply by meljoann** [reply]
OK that worked! Thanks so much @gabek you legend!

But I also had to fix one other thing I did wrong. In the config file, I had
OwncastAddress as the http URL -- it only worked with https
